source/containerimage/pull.go

```diff
@@ -155,14 +157,43 @@ func (p *puller) CacheKey(ctx context.Context, index int) (string, bool, error)
 }
 
 func (p *puller) Snapshot(ctx context.Context) (cache.ImmutableRef, error) {
+	layerTypeWindows := false
+	if platform := p.Puller.Platform; platform != nil {
+		if platform.OS == "windows" && runtime.GOOS != "windows" {
+			ctx = winlayers.UseWindowsLayerMode(ctx)
+			layerTypeWindows = true
```

Comment:
if `platform.OS == "windows" && runtime.GOOS == "windows"` then `layerTypeWindows` is false, which is confusing. I'm trying to think of a better name: `layerNeedsTypeWindows`... Otherwise maybe just add a comment at definition.